go/shared/otel/main.go

```diff
@@ -148,7 +150,15 @@ func InitTracing(ctx context.Context, config *TracingConfig) (err error) {
 	otel.SetTracerProvider(tp)
 
 	var metricExporter *otlpmetricgrpc.Exporter
-	metricExporter, err = otlpmetricgrpc.New(ctx, otlpmetricgrpc.WithInsecure(), otlpmetricgrpc.WithEndpoint(config.Endpoint))
+
+	var metricEndpoint string
+	if config.MetricEndpoint != "" {
+		metricEndpoint = config.MetricEndpoint
+	} else {
+		metricEndpoint = config.TraceEndpoint
+	}
```

Comment:
Do we really need this if we know it won't work? Also, it kinda presumes that `config.TraceEndpoint` is somehow defined when `config.MetricsEndpoint` isn't -- in other words, the `metricsEndpoint` could still end up empty if `config.TraceEndpoint` is also empty. 

Lastly, more of a nit, but if `config.TraceEndpoint` is effectively a default, it's a little less to look at if it was:

```
metricEndpoint := config.TraceEndpoint
if config.MetricEndpoint != "" {
    metricEndpoint = config.MetricEndpoint
}
```

Reply:
> Do we really need this if we know it won't work?

Some services accept both traces and metrics, e.g. the OTEL Collector.

> Also, it kinda presumes that config.TraceEndpoint is somehow defined when config.MetricsEndpoint isn't -- in other words, the metricsEndpoint could still end up empty if config.TraceEndpoint is also empty.

I can add a check for this.